### DISCUSSION TITLE
Controller Typescript

### DISCUSSION THREAD

**Discussion by rusefendi** [react-hook-form/react-hook-form#2879] [Q&A]
Hello!
I have a component which returns <Controller {...props} as={<MyInputComponent />} />
How can I provide correct props typings for my component? Right now I import internal type ControllerProps, but can't make it work.
Basically, the idea is to make props typing work like it working on <Controller /> component, with some additional types.
Thanks!

**Comment by bluebill1049** [comment]
take a look at the TS exmaple in the doc: https://codesandbox.io/s/react-hook-form-v6-controller-ts-4dpm9

**Reply by rusefendi** [reply]
Thanks for the response, I missed Control type, it's useful.
But I wish to have typing for all <Controller /> component, not only the control prop, like name, rules, etc.

**Reply by alvamanu** [reply]
I can't see the Controller type in that codesandbox
